### PR TITLE
storage: use path in WalkDir and Open

### PR DESCRIPTION
**Pull request by glorv**
Signed-off-by: glorv <glorvs@163.com>

### What problem does this PR solve? 
Use file path in ExtrenalStorage method `WalkDir` and `Open`

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has interface methods change

Side effects

Related changes

### Release Note

 - No release note

### CODE REVIEW

File: pkg/storage/local.go [hunk 1]
```diff
@@ -52,13 +52,16 @@ func (l *LocalStorage) WalkDir(ctx context.Context, fn func(string, int64) error
...
 
-// Open a Reader by file name.
-func (l *LocalStorage) Open(ctx context.Context, name string) (ReadSeekCloser, error) {
-	return os.Open(path.Join(l.base, name))
+// Open a Reader by file path.
+func (l *LocalStorage) Open(ctx context.Context, p string) (ReadSeekCloser, error) {
+	if !filepath.IsAbs(p) {
+		p = path.Join(l.base, p)
+	}
+	return os.Open(p)
```
Comment (kennytm) [inline pkg/storage/local.go:64]: this look like a security hole :eyes: why do we support absolute path? (btw use `filepath.Join` instead if `path.Join`.

Comment (glorv) [reply pkg/storage/local.go:64]: the `path` parameter in `filepath.Walk(, func(path, info))`  is absolute path in max osx, and relative path in linux. So I'm not sure if we should be compatible with both

Comment (kennytm) [reply pkg/storage/local.go]: use [`filepath.Rel`](https://golang.org/pkg/path/filepath/#Rel) to force the path to be relative to `l.base`?

Comment (glorv) [reply pkg/storage/local.go]: done

**Pull request merged**